### ISSUE TITLE
build error

### ISSUE THREAD

**Issue by valera39** [Edzelf/ESP32Radio-V2#37]
Hi Ed!
When building in PlatformIO, with the uncommented #define SDCARD function in config.h, a warning appears in the terminal:
Library Manager: Installing SD(esp32) @ ^1.0.5
Warning! Could not find the package with 'SD(esp32) @ ^1.0.5' requirements for your system 'windows_amd64'
then the build fails with the following error:
Compiling .pio\build\esp32\src\main.cpp.o
Compiling .pio\build\esp32\src\utils.cpp.o
Generating partitions .pio\build\esp32\partitions.bin
Compiling .pio\build\esp32\lib1a0\Wire\Wire.cpp.o
Compiling .pio\build\esp32\libc45\PubSubClient\PubSubClient.cpp.o
Compiling .pio\build\esp32\lib274\SPI\SPI.cpp.o
Archiving .pio\build\esp32\lib1a0\libWire.a
Compiling .pio\build\esp32\libcb0\Adafruit BusIO\Adafruit_BusIO_Register.cpp.o
Archiving .pio\build\esp32\lib274\libSPI.a
In file included from src/main.cpp:616:
include/SDcard.h: In function 'void handle_ID3_SD(String&)':
include/SDcard.h:338:13: error: 'SD' was not declared in this scope
   mp3file = SD.open ( path ) ;                              // Open the file
             ^~
include/SDcard.h:338:13: note: suggested alternative: 'SS'
   mp3file = SD.open ( path ) ;                              // Open the file
             ^~
             SS
include/SDcard.h: In function 'bool mount_SDCARD(int8_t)':
include/SDcard.h:458:11: error: 'SD' was not declared in this scope
     if ( !SD.begin ( csPin, SPI, SDSPEED ) )             // Yes, try to init SD card driver
           ^~
include/SDcard.h:458:11: note: suggested alternative: 'SS'
     if ( !SD.begin ( csPin, SPI, SDSPEED ) )             // Yes, try to init SD card driver
           ^~
           SS
include/SDcard.h:464:36: error: 'CARD_NONE' was not declared in this scope
       SD_okay = ( SD.cardType() != CARD_NONE ) ;         // See if known card
                                    ^~~~~~~~~
include/SDcard.h:464:36: note: suggested alternative: 'INADDR_NONE'
       SD_okay = ( SD.cardType() != CARD_NONE ) ;         // See if known card
                                    ^~~~~~~~~
                                    INADDR_NONE
include/SDcard.h: In function 'void scan_SDCARD()':
include/SDcard.h:485:17: error: 'SD' was not declared in this scope
   getsdtracks ( SD, "/", SD_MAXDEPTH ) ;           // Build file list
                 ^~
include/SDcard.h:485:17: note: suggested alternative: 'SS'
   getsdtracks ( SD, "/", SD_MAXDEPTH ) ;           // Build file list
                 ^~
                 SS
Compiling .pio\build\esp32\libcb0\Adafruit BusIO\Adafruit_I2CDevice.cpp.o
Archiving .pio\build\esp32\libc45\libPubSubClient.a
Compiling .pio\build\esp32\libcb0\Adafruit BusIO\Adafruit_SPIDevice.cpp.o
*** [.pio\build\esp32\src\main.cpp.o] Error 1
How can this be fixed?

**Comment by Edzelf** [comment]
SD is still experimental.
Try to add:
"lib_ignore = SD"
to your platformio.ini

**Comment by valera39** [comment]
Thanks for the quick response. Unfortunately, this did not help, the build still fails.

**Comment by Edzelf** [comment]
Try the new version of platformio.ini.

**Comment by valera39** [comment]
i have the latest version PlatformIO  Core 5.2.5·Home 3.4.1

**Comment by Edzelf** [comment]
I mean the file platformio.ini on the ESPRadio github.

**Comment by valera39** [comment]
Replaced platformio.ini, Now like this: 
In file included from src/main.cpp:616:
include/SDcard.h:36:12: fatal error: SD.h: No such file or directory

************************************************************
* Looking for SD.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:SD.h"
* Web  > https://registry.platformio.org/search?q=header:SD.h
*
************************************************************

   #include <SD.h>
            ^~~~~~
compilation terminated.
*** [.pio\build\esp32\src\main.cpp.o] Error 1

**Comment by Edzelf** [comment]
I just downloaded the release version of ESP32Radio-V2 on a different test machine.
I copied your config.h file.
Compiled okay.
So it is a big mystery.
My SD.h is located at:
C:\Users\ed\.platformio\packages\framework-arduinoespressif32\libraries\SD\src\SD.h

**Comment by Edzelf** [comment]
Bingo!  After update of pio, I got SD compile errors.  Now I have something to look for.

**Comment by valera39** [comment]
Yes, it's definitely a mystery. Thanks for the answers, I'll try again in the evening on another computer.
Have a nice day!

**Comment by Edzelf** [comment]
I included the SD library to the "lib" directory.  That seems to cure the compilation error.  Will test the result later.

**Comment by valera39** [comment]
I didn't wait for the evening :)
Everything came together without errors, thanks for the excellent work! But it looks like my ESP32 has failed so I can't download and test it yet.
How to add Cyrillic support?
Have a nice day!
